Sources/Basic/OSLog.swift

```diff
@@ -0,0 +1,248 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2018 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+#if canImport(os)
+import os.log
+import os.signpost
+#endif
+
+// MARK :- OSLog APIs
+
+/// A custom log object that can be passed to logging functions in order to send
+/// messages to the logging system.
+///
+/// This is a thin wrapper for Darwin's log APIs to make them usable without
+/// platform and availability checks.
+public final class OSLog {
+
+    private let __log: Any?
+
+  #if canImport(os)
+    @available(OSX 10.12, *)
+    @usableFromInline var _log: os.OSLog {
+        return __log as! os.OSLog
+    }
+  #endif
+
+    private init(_ __log: Any? = nil) {
+        self.__log = __log
+    }
+
+    /// Creates a custom log object.
+    public convenience init(subsystem: String, category: String) {
+      #if canImport(os)
+        if #available(OSX 10.12, *) {
+            self.init(os.OSLog(subsystem: subsystem, category: category))
+        } else {
+            self.init(nil)
+        }
+      #else
+        self.init(nil)
+      #endif
+    }
+
+    /// The shared default log.
+    public static let disabled: OSLog = {
+      #if canImport(os)
+        if #available(OSX 10.12, *) {
+            return OSLog(os.OSLog.disabled)
+        } else {
+            return OSLog(nil)
+        }
+      #else
+        return OSLog(nil)
+      #endif
+    }()
+
+    /// The shared disabled log.
+    public static let `default`: OSLog = {
+      #if canImport(os)
+        if #available(OSX 10.12, *) {
+            return OSLog(os.OSLog.default)
+        } else {
+            return OSLog(nil)
+        }
+      #else
+        return OSLog(nil)
+      #endif
+    }()
+}
+
+/// Sends a message to the logging system.
+///
+/// This is a thin wrapper for Darwin's log APIs to make them usable without
+/// platform and availability checks.
+@inlinable public func os_log(log: OSLog = .default, _ message: StaticString, _ args: CVarArg...) {
+  #if canImport(os)
+    if #available(OSX 10.14, *) {
```

Comment:
Yes, that's true.

https://developer.apple.com/documentation/os/os_log

Reply:
Shouldn't it also be `macOS` instead of `OSX`? I think it would also be cleaner to copy the availability attributes from `OSLog` itself (with all OSLog-supported platforms for completeness, even if SwiftPM is not supported).

Reply:
🤷‍♂️